content/en/docs/instrumentation/net/getting-started.md

```diff
@@ -1,206 +1,261 @@
-through `System.Diagnostics` and the greater OpenTelemetry ecosystem, such as
-OpenTelemetry Protocol (OTLP) and the OpenTelemetry Collector.
+This page will show you how to get started with OpenTelemetry in .NET.
 
-## ASP.NET Core
+You will learn how you can instrument a simple .NET application automatically,
+in such a way that [traces][], [metrics][] and [logs][] are emitted to the
+console.
 
-The following example demonstrates using Instrumentation Libraries and manual
-instrumentation via an ASP.NET Core app.
+## Prerequisites
 
-First, create your basic ASP.NET Core site:
+Ensure that you have the following installed locally:
 
-```shell
-dotnet new mvc
-```
+- [.NET SDK](https://dotnet.microsoft.com/download/dotnet) 6+
 
-Next, Add the Core OpenTelemetry packages
+## Example Application
 
-```shell
-dotnet add package OpenTelemetry.Exporter.Console
-dotnet add package OpenTelemetry.Extensions.Hosting
-```
+The following example uses a basic
+[Minimal API with ASP.NET Core](https://learn.microsoft.com/aspnet/core/tutorials/min-web-api)
+application. If you are not using ASP.NET Core, that's OK — you can still use
+OpenTelemetry .NET Automatic Instrumentation.
 
-Now let's add the instrumentation packages for ASP.NET Core. This will give us
-some automatic spans for each HTTP request to our app.
+For more elaborate examples, see
+[examples](/docs/instrumentation/net/examples/).
 
-```shell
-dotnet add package OpenTelemetry.Instrumentation.AspNetCore --prerelease
-```
+### Create and launch an HTTP Server
 
-_Note that as the Semantic Conventions for attribute names are not currently
-stable the instrumentation package is currently not in a released state. That
-doesn't mean that the functionality itself is not stable. This means that you
-need to use the `--prerelease` flag, or install a specific version of the
-package_
+To begin, set up an environment in a new directory called `dotnet-simple`.
+Within that directory, execute following command:
 
-### Setup
+```sh
+dotnet new web
+```
 
-Next, we need to add OpenTelemetry to our Service Collection in `Program.cs` so
-that its listening correctly.
+In the same directory, replace the content of `Program.cs` with the following
+code:
 
 ```csharp
-using System.Diagnostics;
-using OpenTelemetry.Resources;
-using OpenTelemetry.Trace;
+using System.Globalization;
 
 var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
 
-// .. other setup
+var logger = app.Logger;
+var random = new Random();
 
-builder.Services.AddOpenTelemetry()
-    .WithTracing(tracerProviderBuilder =>
-        tracerProviderBuilder
-            .AddSource(DiagnosticsConfig.ActivitySource.Name)
-            .ConfigureResource(resource => resource
-                .AddService(DiagnosticsConfig.ServiceName))
-            .AddAspNetCoreInstrumentation()
-            .AddConsoleExporter());
+app.MapGet("/rolldice/{player?}", HandleRollDice);
 
-// ... other setup
+app.Run();
+return;
```

Comment:
Mmmm, I don't really like this. I'd much prefer either:

1. The handler moves up pass `app.Run()`
2. The handler defined inline
3. The handler in another file

I think it's just a little too subtle to have the `return` statement there.

Reply:
What do you think?

Reply:
I have started with your option 1:
```csharp
using System.Globalization;

var builder = WebApplication.CreateBuilder(args);
var app = builder.Build();

var logger = app.Logger;
var random = new Random();

int RollDice()
{
    return random.Next(1, 7);
}

string HandleRollDice(string? player)
{
    var result = RollDice();

    if (string.IsNullOrEmpty(player))
    {
        logger.LogInformation("Anonymous player is rolling the dice: {result}", result);
    }
    else
    {
        logger.LogInformation("{player} is rolling the dice: {result}", player, result);
    }

    return result.ToString(CultureInfo.InvariantCulture);
}

app.MapGet("/rolldice/{player?}", HandleRollDice);

app.Run();
```

Then checked all suggestion in IDE (VS2023+R# with default settings) and applied all reformat suggestions.
It leads to the current state (it was mainly fixing `Put local function '{name}' after 'return' statement`).

I am fine with both solutions. If you want, I can add commit with this version. Code is simple enough in both cases.

I would avoid creating separate file or inlining methods. I think one file with some logical separation is good compromise for this kind of examples.

Reply:
Mmmm, I should ping a friend on the team to complain to them about that refactoring :)

Yeah I'd prefer Option 1 if possible -- I think it's fine for people to use R# refactors on their own, but I do think it makes the code more subtle to understand, and we want things to be as straightforward as possible on the docs site (I also acknowledge that "straightforward" can be subjective!)